sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/binaryfile/BinaryFileFormatSuite.scala

```diff
@@ -140,4 +195,45 @@ class BinaryFileFormatSuite extends QueryTest with SharedSQLContext with SQLTest
     }
   }
 
+
+  def testCreateFilterFunctions(
```

Comment:
See comment above. We can just unit test the `buildReader` func, which would look similar to your test here.